krun/util.py

```diff
@@ -332,3 +333,23 @@ def get_git_version():
         run_shell_cmd("sh -c 'cd %s && git rev-parse --verify HEAD'" % DIR)
 
     return out.strip()  # returns the hash
+
+
+def get_instr_json_dir(config):
+    assert config.filename.endswith(".krun")
+    config_base = config.filename[:-5]
+    return os.path.join(os.getcwd(), "%s_instr_data" % config_base)
+
+
+def dump_instr_json(key, exec_num, config, instr_data):
+    """Write per-execution instrumentation data to a separate JSON file"""
+
+    instr_json_dir = get_instr_json_dir(config)
+    if not os.path.exists(instr_json_dir):
+        os.mkdir(instr_json_dir)
+
+    filename = "%s__%s.json.bz2" % (key.replace(":", "_"), exec_num)
+    path = os.path.join(instr_json_dir, filename)
+    assert not os.path.exists(path)
```

Comment:
Yes, I understand :) I'm just saying add a comment explaining *where* that test happens, otherwise future readers will have the same initial comment I did.